src/extractor/guidance/node_based_graph_walker.cpp

```diff
@@ -167,6 +168,37 @@ operator()(const NodeID /*nid*/,
          angularDeviation(min_element->angle, STRAIGHT_ANGLE) < NARROW_TURN_ANGLE) &&
         angularDeviation(initial_bearing, min_element->bearing) < NARROW_TURN_ANGLE;
 
+    // do not allow major turns in the road, if other similar turns are present
+    // e.g.a turn at the end of the road:
+    //
+    // a - - a - - a - b - b
+    //             |
+    //       a - - a
+    //             |
+    //             c
+    //
+    // Such a turn can never be part of a merge
+    // We check if there is a similar turn to the other side. If such a turn exists, we consider the
+    // continuation of the road not possible
+    if (util::angularDeviation(STRAIGHT_ANGLE, min_element->angle) > GROUP_ANGLE)
+    {
+        auto deviation = util::angularDeviation(STRAIGHT_ANGLE, min_element->angle);
+        auto opposite_angle = min_element->angle >= STRAIGHT_ANGLE ? (STRAIGHT_ANGLE - deviation)
+                                                                   : (STRAIGHT_ANGLE + deviation);
+        auto opposite = intersection.findClosestTurn(opposite_angle);
+        auto opposite_deviation = util::angularDeviation(STRAIGHT_ANGLE, opposite->angle);
+        if (opposite_deviation <= deviation || (deviation / opposite_deviation) < 1.5)
```

Comment:
don't understand the condition because it is always true:
```
opposite_deviation <= deviation or deviation / opposite_deviation < 1.5
1 <= deviation / opposite_deviation  or deviation / opposite_deviation < 1.5
deviation / opposite_deviation >= 1 or deviation / opposite_deviation < 1.5
deviation / opposite_deviation in R
```

it  makes the next lines uncovered https://codecov.io/gh/Project-OSRM/osrm-backend/compare/37774a331a7b4a7f839b06ed4d281eb192bf90cf...df733835f9189dcf232c731d9f2a35d806db92bc/diff#D1-195...198

Reply:
Thats actually a bug and should be the other way around. I want to divide by deviation (first condition ensuring that deviation > 0). Great catch!